zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanTask.java

```diff
@@ -204,16 +204,18 @@ public void run() {
                 }
             } else {
                 logger.error(
-                        "Parser failed on record {} from History table", href.getHistoryId(), e);
+                        "Parser failed on record {} from History table (perhaps it was deleted between when it was enqueued and now)",
+                        href.getHistoryId());
+                logger.debug(e);
                 HttpMessage msg;
                 try {
                     msg = href.getHttpMessage();
                     logger.error(
                             "Parser failed on record {} from History table, msg {}",
                             href.getHistoryId(),
-                            msg,
-                            e);
-                    logger.error("Req Header {}", msg.getRequestHeader(), e);
+                            msg);
+                    logger.debug(e);
+                    logger.debug("Req Header {}", msg.getRequestHeader());
```

Comment:
Does this work as intended?

Reply:
This part may be unnecessary in the context of the original issue. I was going for consistency. Will test more.